src/charm.py

```diff
@@ -1,104 +1,75 @@
-        """Define and start a workload using the Pebble API.
-
-        TEMPLATE-TODO: change this example to suit your needs.
-        You'll need to specify the right entrypoint and environment
-        configuration for your specific workload. Tip: you can see the
-        standard entrypoint of an existing container using docker inspect
-
-        Learn more about Pebble layers at https://github.com/canonical/pebble
-        """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": self.model.config["thing"]},
-                }
-            },
-        }
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", pebble_layer, combine=True)
-        # Autostart any services that were defined with startup: enabled
-        container.autostart()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
+        self.cloud_config = CloudConfigRequires(self)
+        self.lk_helpers = LightKubeHelpers(self)
+        self.framework.observe(self.on.install, self._install_or_upgrade)
+        self.framework.observe(self.on.upgrade_charm, self._install_or_upgrade)
+        self.framework.observe(self.on.cloud_config_relation_created, self._install_or_upgrade)
+        self.framework.observe(self.cloud_config.on.ready, self._install_or_upgrade)
+        self.framework.observe(self.on.leader_elected, self._set_version)
+        self.framework.observe(self.on.stop, self._cleanup)
+
+    def _install_or_upgrade(self, event):
+        if not self.cloud_config.relations:
+            self.unit.status = BlockedStatus("Missing cloud-config relation")
+            return
+        if not self.cloud_config.is_ready():
+            self.unit.status = WaitingStatus("Waiting for cloud-config")
+            return
+        self.unit.status = MaintenanceStatus("Deploying Cinder CSI")
+        for manifest in self.manifests.glob("**/*.yaml"):
+            if "secret" in manifest.name:
+                # The upstream secret contains dummy data, so skip it.
+                continue
+            self.lk_helpers.apply_manifest(manifest)
+        self.lk_helpers.apply_resource(
+            StorageClass,
+            name="cinder",
+            provisioner="cinder.csi.openstack.org",
+            annotations={"juju.io/workload-storage": "true"},
+        )
         self.unit.status = ActiveStatus()
```

Comment:
Should there be ANYTHING here?  How about the git version of the charm running?  I dunno.  something

```suggestion
        self.unit.status = ActiveStatus("Ready")
```

Reply:
See https://github.com/canonical/operator/issues/126. The direction I received was that active status should stand on its own, and that eventually, we'd get the ability for charms to provide structured output for any additional info.